build.rs

```diff
@@ -0,0 +1,7 @@
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    println!("cargo::rerun-if-changed=build.rs");
+    match std::env::var("CARGO_CFG_TARGET_OS") {
+        Ok(val) if val == "linux" => Ok(()),
+        _ => Err("Building for anything but Linux is not supported by socketcan".into()),
+    }
+}
```

Comment:
This works, but I usually force a compile error for an error message that's even more clear:

```
// build.rs

#[cfg(not(target_os = "linux"))]
compile_error!("socketcan can only be build for a Linux target");

fn main() {}
```
which gives:

```
error: socketcan can only be build for a Linux target
 --> build.rs:4:1
  |
4 | compile_error!("socketcan can only be build for a Linux target");
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: could not compile `buildtgt` (build script) due to 1 previous error
```

Reply:
This does not work for cross compiles, as far as I know. Inside a build script, you cannot rely on #[cfg] for target_os but have to use the cargo environment variable. The reason is that the build script is built for the architecture that you're running the build on because it gets executed there.

Reply:
Oh, yeah, I think you're right. This keeps tripping me up.